### PR TITLE
docs: Add example for handling webhook batch requests

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR adds an example to our docs of how to handle batch requests in a webhook source, using `jsonb_array_elements(...)` function.

### Motivation

There are users that need to increase their throughput for webhook sources, an easy way to do that is to batch requests. We want to add docs to help them get this setup.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Adds docs for handling batch requests with webhooks

### CODE REVIEW

File: doc/user/content/sql/create-source/webhook.md [hunk 1]
```diff
@@ -254,6 +254,25 @@ If the feature is enabled, when casting from `text` to `timestamp` you should pr
 
 {{< /note >}}
 
+### Handling batch events
+
+The application pushing events to your webhook source, may batch multiple events into a single
```
Comment (chaas) [inline doc/user/content/sql/create-source/webhook.md:259]: just a nit
```suggestion
The application pushing events to your webhook source may batch multiple events into a single
```

File: doc/user/content/sql/create-source/webhook.md [hunk 2]
```diff
@@ -254,6 +254,25 @@ If the feature is enabled, when casting from `text` to `timestamp` you should pr
...
+HTTP request. You can automatically expand this batch of requests into separate rows, using a
+[`MATERIALIZED VIEW`](/sql/create-materialized-view/) and the [`jsonb_array_elements`](/sql/types/jsonb/#functions)
+function.
+
+```sql
+-- Webhook source that parses request bodies as JSON.
+CREATE SOURCE webhook_source_json_batch IN CLUSTER my_cluster FROM WEBHOOK
+  BODY FORMAT JSON
+  INCLUDE HEADERS
```
Comment (chaas) [inline doc/user/content/sql/create-source/webhook.md:268]: nit: do you need a semicolon after this statement and the below one?

File: doc/user/content/sql/create-source/webhook.md [hunk 3]
```diff
@@ -254,6 +254,25 @@ If the feature is enabled, when casting from `text` to `timestamp` you should pr
 
 {{< /note >}}
 
+### Handling batch events
+
+The application pushing events to your webhook source, may batch multiple events into a single
+HTTP request. You can automatically expand this batch of requests into separate rows, using a
+[`MATERIALIZED VIEW`](/sql/create-materialized-view/) and the [`jsonb_array_elements`](/sql/types/jsonb/#functions)
```
Comment (chaas) [inline doc/user/content/sql/create-source/webhook.md:261]: Is it better to be a materialized view or just a regular view? What's the best practice on creating intermediary mat views vs using regular views?

**Review by chaas — APPROVED** [review]
This is great!

File: doc/user/content/sql/create-source/webhook.md (hunk 2 above)
Comment (ParkMyCar) [reply doc/user/content/sql/create-source/webhook.md]: Done!

File: doc/user/content/sql/create-source/webhook.md (hunk 1 above)
Comment (ParkMyCar) [reply doc/user/content/sql/create-source/webhook.md]: Done!

**Pull request merged**